libsplinter/src/oauth/store/diesel/mod.rs

```diff
@@ -0,0 +1,209 @@
+impl InflightOAuthRequestStore
+    for DieselInflightOAuthRequestStore<diesel::sqlite::SqliteConnection>
+{
+    fn insert_request(
+        &self,
+        request_id: String,
+        pending_authorization: PendingAuthorization,
+    ) -> Result<(), InflightOAuthRequestStoreError> {
+        let connection = self
+            .connection_pool
+            .get()
+            .map_err(|err| InternalError::from_source(Box::new(err)))?;
+        InflightOAuthRequestOperations::new(&*connection).insert_request(
+            models::OAuthInflightRequest {
+                id: request_id,
+                pkce_verifier: pending_authorization.pkce_verifier,
+                client_redirect_url: pending_authorization.client_redirect_url,
+            },
+        )
+    }
+
+    fn remove_request(
+        &self,
+        request_id: &str,
+    ) -> Result<Option<PendingAuthorization>, InflightOAuthRequestStoreError> {
+        let connection = self
+            .connection_pool
+            .get()
+            .map_err(|err| InternalError::from_source(Box::new(err)))?;
+        InflightOAuthRequestOperations::new(&*connection)
+            .remove_request(request_id)
+            .map(|opt_request| opt_request.map(PendingAuthorization::from))
+    }
+
+    fn clone_box(&self) -> Box<dyn InflightOAuthRequestStore> {
+        Box::new(Self {
+            connection_pool: self.connection_pool.clone(),
+        })
+    }
+}
+
+#[cfg(feature = "oauth-inflight-request-store-postgres")]
+impl InflightOAuthRequestStore for DieselInflightOAuthRequestStore<diesel::pg::PgConnection> {
+    fn insert_request(
+        &self,
+        request_id: String,
+        pending_authorization: PendingAuthorization,
+    ) -> Result<(), InflightOAuthRequestStoreError> {
+        let connection = self
+            .connection_pool
+            .get()
+            .map_err(|err| InternalError::from_source(Box::new(err)))?;
+        InflightOAuthRequestOperations::new(&*connection).insert_request(
+            models::OAuthInflightRequest {
+                id: request_id,
+                pkce_verifier: pending_authorization.pkce_verifier,
+                client_redirect_url: pending_authorization.client_redirect_url,
+            },
+        )
+    }
+
+    fn remove_request(
+        &self,
+        request_id: &str,
+    ) -> Result<Option<PendingAuthorization>, InflightOAuthRequestStoreError> {
+        let connection = self
+            .connection_pool
+            .get()
+            .map_err(|err| InternalError::from_source(Box::new(err)))?;
+        InflightOAuthRequestOperations::new(&*connection)
+            .remove_request(request_id)
+            .map(|opt_request| opt_request.map(PendingAuthorization::from))
+    }
+
+    fn clone_box(&self) -> Box<dyn InflightOAuthRequestStore> {
+        Box::new(Self {
+            connection_pool: self.connection_pool.clone(),
+        })
+    }
+}
+
+impl From<models::OAuthInflightRequest> for PendingAuthorization {
```

Comment:
In other stores we've put all `From` implementations in the store's top-level error module to keep all error boilerplate in one place.

Reply:
This isn't an error conversion, it converts the model struct to the one the oauth client uses.

Reply:
Oops, meant to put this on the error conversions below